src/main/java/io/github/moulberry/notenoughupdates/listener/NEUEventListener.java

```diff
@@ -78,80 +80,69 @@ public NEUEventListener(NotEnoughUpdates neu) {
-							first_len = len;
-						}
-						int missing_len = first_len - len;
-						if (missing_len > 0) {
-							StringBuilder sb = new StringBuilder(line);
-							for (int i = 0; i < missing_len / 8; i++) {
-								sb.insert(0, " ");
-							}
-							line = sb.toString();
-						}
-						line = line.replaceAll("\\{version}", version);
-						Minecraft.getMinecraft().thePlayer.addChatMessage(new ChatComponentText(line));
-					}
-
-					neu.displayLinks(o);
-
-					Minecraft.getMinecraft().thePlayer.addChatMessage(new ChatComponentText(""));
-				} else if (shouldPreUpdate && NotEnoughUpdates.VERSION_ID == o.get("version").getAsInt()) {
-					String pre_update_msg = o.get("pre_update_msg").getAsString();
-
-					int first_len = -1;
-					for (String line : pre_update_msg.split("\n")) {
-						FontRenderer fr = Minecraft.getMinecraft().fontRendererObj;
-						int len = fr.getStringWidth(line);
-						if (first_len == -1) {
-							first_len = len;
-						}
-						int missing_len = first_len - len;
-						if (missing_len > 0) {
-							StringBuilder sb = new StringBuilder(line);
-							for (int i = 0; i < missing_len / 8; i++) {
-								sb.insert(0, " ");
-							}
-							line = sb.toString();
-						}
-						line = line.replaceAll("\\{pre_version}", preVersion);
-						Minecraft.getMinecraft().thePlayer.addChatMessage(new ChatComponentText(line));
-					}
-
-					neu.displayLinks(o);
-
-					Minecraft.getMinecraft().thePlayer.addChatMessage(new ChatComponentText(""));
+				int fullReleaseVersion =
+					o.has("version_id") && o.get("version_id").isJsonPrimitive() ? o.get("version_id").getAsInt() : -1;
+				int preReleaseVersion =
+					o.has("version_id") && o.get("version_id").isJsonPrimitive() ? o.get("version_id").getAsInt() : -1;
+				int hotfixVersion =
+					o.has("hotfix_id") && o.get("hotfix_id").isJsonPrimitive() ? o.get("hotfix_id").getAsInt() : -1;
+
+				boolean hasFullReleaseAvailableForUpgrade = fullReleaseVersion > NotEnoughUpdates.VERSION_ID;
+				boolean hasHotfixAvailableForUpgrade =
+					fullReleaseVersion == NotEnoughUpdates.VERSION_ID && hotfixVersion > NotEnoughUpdates.HOTFIX_VERSION_ID;
+				boolean hasPreReleaseAvailableForUpdate =
+					fullReleaseVersion == NotEnoughUpdates.VERSION_ID && preReleaseVersion > NotEnoughUpdates.PRE_VERSION_ID;
+
+				int updateChannel = NotEnoughUpdates.INSTANCE.config.notifications.showUpdateMsg; /* 1 = Full, 2 = Pre */
+				if (hasFullReleaseAvailableForUpgrade || (hasHotfixAvailableForUpgrade && updateChannel == 1)) {
+					displayUpdateMessage(o, o.get("update_msg").getAsString(), o.get("update_link").getAsString());
+				} else if (hasPreReleaseAvailableForUpdate && updateChannel == 2) {
+					displayUpdateMessage(o, o.get("pre_update_msg").getAsString(), o.get("pre_update_link").getAsString());
 				}
-			} catch (Exception ignored) {
+			} catch (Exception e) {
+				e.printStackTrace();
+				Minecraft.getMinecraft().thePlayer.addChatMessage(new ChatComponentText(
```

Comment:
this is gonna trigger every time someone doesn't have the repo? i don't think the message is worthwhile

Reply:
this try isnt even entered unless `repo.exists()`